core/src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java

```diff
@@ -974,6 +978,55 @@ public JsonNode call(JsonNode input, JsonNode[] arguments) {
     }
   }
 
+  // ===== UUID
+
+  public static class Uuid extends AbstractFunction {
+
+    public Uuid() {
+      super("uuid", 0, 2);
+    }
+
+    private long maskMSB(long number) {
+      final long version = 1 << 12;
+      long least12SignificantBit = (number & 0x000000000000FFFFL) >> 4;
+      return (number & 0xFFFFFFFFFFFF0000L) + version + least12SignificantBit;
+    }
+
+    private long maskLSB(long number) {
+      final long LSB_MASK = 0x3FFFFFFFFFFFFFFFL;
+      final long LSB_VARIANT3_BITFLAG = 0x8000000000000000L;
+      return (number & LSB_MASK) + LSB_VARIANT3_BITFLAG;
+    }
+
+    public JsonNode call(JsonNode input, JsonNode[] arguments) {
+      String uuid;
+      if (arguments.length == 0) {
+        uuid = UUID.randomUUID().toString();
+      } else if (arguments.length == 1) {
+        long msb;
+        if (arguments[0].isNull()) {
+          msb = 0;
```

Comment:
Why are we treating `null` as `0` here? Could we not just simply omit this part?

Reply:
removed whole branch handling only 1 parameter